### PR TITLE
Update GitHub Actions to use @v4 to prevent deprecation issues

### PR DESCRIPTION
**Pull request by Yerzhaisang**
### Description
- Updated all workflow files to replace @v3 with @v4
- Ensured artifact actions comply with GitHub's deprecation policy
- Added Dependabot configuration for automatic GitHub Actions updates
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-py-ml/issues/426
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Yerzhaisang** [comment]
All workflows have been successfully upgraded to use @v4 for their GitHub Actions. This ensures compatibility with GitHub’s deprecation of @v3 versions, improving security and future-proofing the workflows.

**Exception**:
- dangoslen/changelog-enforcer remains at @v3 because the latest available version is v3, and no v4 release currently exists.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,10 +1,15 @@
+version: 2
 updates:
-  - directory: /
-    open-pull-requests-limit: 3
-    package-ecosystem: pip
+  - package-ecosystem: "pip"
```
Comment (dhrubo-os) [inline .github/dependabot.yml:3]: why do we need to change this? `pip` to `"pip"` ?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-      interval: weekly
+      interval: "weekly"
+    open-pull-requests-limit: 3
     labels:
       - "dependabot"
       - "dependencies"
-version: 2
\ No newline at end of file
+      
+  - package-ecosystem: "github-actions"
```
Comment (dhrubo-os) [inline .github/dependabot.yml:12]: why do we need this?

**Review by dhrubo-os — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**